tests/tools/test_array_readwriter.py

```diff
@@ -0,0 +1,23 @@
+import tempfile
```

Comment:
how about renaming the test file into `test_utils.py` as we may add more methods into `utils` and more testcases into this file?

Reply:
Sure.